### ISSUE TITLE
[feature] length-or op in addition to ||=, //=

### ISSUE THREAD

**Issue by atoomic** [Perl/perl5#17191]
this is a discussion started by @FGasper from https://github.com/p5h/p5summit-2019/issues/25

Over the years I’ve not infrequently encountered this:
```
$form_value ||= $default
```
.. which undesirably rejects 0 as input.

Defined-or doesn’t help here because the form submits regardless.

It would be very helpful if Perl (7?) added a “length-or”: undef and q// are falsy, but everything else is truthy.

we already have
```
||=
//=
```
what about something like `//=` or any other ?

Feedback from @leonerd 

In a sense this points out the duality of the string-or-number model of Perl's type system in that is demonstrates the question of how you ask the "truth" of a value really depends on whether you wanted the stringy truth or the numerical truth.

```perl
# stringily false
undef, ""

# numerically false
undef, 0,  # plus "" which is exempt from number->string conversion warnings
```

**Comment by xsawyerx** [comment]
My view of this is that we're getting into more and more nuanced, specific cases and introducing new syntax for them. I'm generally not in favor of this perspective.

**Comment by rsFalse** [comment]
> this is a discussion started by @FGasper from [p5h/p5summit-2019#25](https://github.com/p5h/p5summit-2019/issues/25)

> 
> It would be very helpful if Perl (7?) added a “length-or”: undef and q// are falsy, but everything else is truthy.
> 
> we already have
> 
> ```
> ||=
> //=
> ```
> 
> what about something like `//=` or any other ?

These examples of existing operators are combined operators (OR + assignment). We can try to find a suitable symbols for both: _length-or_ and _length-or-assignment_.

I'm not sure if this idea is good to introduce such operator. 

Here is an example how it may look: `|.|` and `|.|=` or `/./` and `/./=`. It resembles a match operator `"a" =~ m{.}s`, but similar appearance also can annoy. And also it looks too similar to new `|.=` bitwise string OR operator.

**Comment by Leont** [comment]
> My view of this is that we're getting into more and more nuanced, specific cases and introducing new syntax for them. I'm generally not in favor of this perspective.

I agree. We should only add shortened syntax for things that can already be expressed with more words if the actions is common, and I don't think this is nearly common enough.
